pkg/plugins/generic/generic_plugin.go

```diff
@@ -217,35 +269,35 @@ func needDrainNode(desired sriovnetworkv1.Interfaces, current sriovnetworkv1.Int
 	return
 }
 
-func needRebootNode(state *sriovnetworkv1.SriovNetworkNodeState, loadVfioDriver *uint, loadVirtioVdpaDriver *uint) (needReboot bool) {
-	needReboot = false
-	if *loadVfioDriver != loaded {
-		if needVfioDriver(state) {
-			*loadVfioDriver = loading
-			update, err := tryEnableIommuInKernelArgs()
+func nodeToRebootIfVfio(state *sriovnetworkv1.SriovNetworkNodeState, driverMap DriverStateMapType) (updateNode bool) {
+	updateNode = false
+	for driverID, driverState := range driverMap {
```

Comment:
driverMap is a .... map.

no need to iterate over it. instead just index it by `Vfio`